core/src/env_platform/db_handler.bal

```diff
@@ -543,3 +544,166 @@ function isMinistryHasUser(string ministryId, string userId) returns boolean|err
         return false;
     }
 }
+
+# The `postCommentInApplication` function will append the comment to the existing 
+# comment array of the application with the help of the given application ID.
+# 
+# + applicationId - ID of the application.
+# + message - Message to be added.  
+# + return - This function will return a boolean indicating whether the message is added or an appropriate error.
+function postCommentInApplication(string applicationId, Message message) returns boolean|error {
+
+    map<json>[] applications = check applicationCollection->find({"applicationId": applicationId, status: "submit"});
+    if (applications.length() == 0) {
+        return error("Not found", message = "Application is not found with ID " + applicationId + ".");
+    } else {
+        map<json>|error application = <map<json>>applications[0];
+        if (application is error) {
+            return error("Not found", message = "Application is not found with ID " + applicationId + ".");
+        } else {
+            if (application.comments is error) {
+                map<json> comment = constructComment(message, 0);
+                log:printDebug("The constructed comment is " + comment.toString() + ".");
+                int updated = check applicationCollection->update({comments: [comment]},
+                    {"applicationId": applicationId});
+                return updated == 1 ? true : false;
+            } else {
+                json[] comments = check trap <json[]>application.comments;
+                map<json> comment = constructComment(message, comments.length());
+                log:printDebug("The constructed comment is " + comment.toString() + ".");
+                comments.push(comment);
+                int updated = check applicationCollection->update({comments: comments},
+                    {"applicationId": applicationId});
+                return updated == 1 ? true : false;
+            }
+        }
+    }
+}
+
+# The `isApplicationRelatedTo` function will check whether the given application is related 
+# to the user.
+# 
+# + userType - Type of the user.
+# + userId - ID of the user.
+# + applicationId - ID of the application.
+# + return - This will return a boolean indicating whether the application is related 
+# to the user or an appropriate error.
+function isApplicationRelatedTo(string userType, string userId, string applicationId) returns boolean|error {
```

Comment:
Is it possible to have debug logs here as well?

Reply:
Added with new commit